### DISCUSSION TITLE
xarray: How to get the coordinates of the maximum and minimum value over a spatial domain in xarray?

### DISCUSSION THREAD

**Discussion by anubhavchoudhary** [pydata/xarray#8036] [Q&A]
I have a netcdf file containing a surface variable over a latlon area. I want to get the coordinate (as lat, lon value) of maximum as well as minimum value of that variable over a certain area which I can define with latlon box. How to do it in xarray ?

**Comment by TomNicholas** [comment]
There might be a better way, but I think this is one way:

First mask out all the data that isn't in your lat/long box. If your latitude and longitude are 1D then you could use `.sel()`, otherwise use `.where()`. You have two conditions (one for latitude and one for longitude), so combine them with `np.logical_and()`:

```python
condition = np.logical_and(data.lat > 10.0, data.lon < 50.0)
box = data.where(condition)
```

Then use `DataArray.argmax()` across all the dimensions of the result to find the indices of the maximum values. 

```python
inds_of_max = box.argmax(dims=['lat', 'lon'])
```

Pass those indices to your latitude/longitude coordinate arrays to get the (lat, lon) pair you want.

```python
lat, lon = data.lat[inds_of_max], data.lon[inds_of_max]
```

(This would probably be better placed on stackoverflow than here but doesn't matter)

**Comment by shoyer** [comment]
In fact, this was already asked on StackOverflow :)
https://stackoverflow.com/questions/40179593/how-to-get-the-coordinates-of-the-maximum-in-xarray

I'm going to close this, but feel free to comment or reopen if you have any further questions.
